packages/testkit-backend/src/channel/socket.js

```diff
@@ -0,0 +1,83 @@
+import Channel from './interface'
+import net from 'net'
+import { randomBytes } from 'crypto'
+import Protocol from './testkit-protocol'
+
+function generateRandomId () {
+  return randomBytes(16).toString('hex')
+}
```

Comment:
Why a random ID instead of incrementing a counter? It could well be that I didn't understand the structure of this change. I thought there is exactly one channel per backend and TestKit should only ever start one backend.

Reply:
The random id is to not have any global state in responsible for generate id. It's a bit of preciosity from my side.

About the support for more than one Testkit (client) being connect, It was already something we supported in the original code. 

For each client connect, the backend started it own protocol parser and request handler context. However, now this behaviour is really clear because it's not hidden in a lot of code doing different things. 

I'd like to keep this support for multiple clients at same time, It makes things well segregated and enables the backend run tests in parallel.